daemon/logger/jsonfile/jsonfile.go

```diff
@@ -47,23 +52,54 @@ func Init(info logger.Info) (logger.LogDriver, error) {
 		}
 	}
 
-	return NewJSONLogFile(logPath, 0644, func(msg *logger.LogMessage) ([]byte, error) {
+	return NewJSONLogFile(logPath, 0644, info.LogConfig, func(msg *logger.LogMessage) ([]byte, error) {
 		return Marshal(msg, extra)
 	})
 }
 
 // NewJSONLogFile returns new JSONLogFile instance.
-func NewJSONLogFile(logPath string, perms os.FileMode, marshalFunc MarshalFunc) (*JSONLogFile, error) {
+func NewJSONLogFile(logPath string, perms os.FileMode, logConfig map[string]string, marshalFunc MarshalFunc) (*JSONLogFile, error) {
+	var (
+		currentsize uint64
+		maxsize     uint64
+		maxfiles    = 1
+	)
 	f, err := os.OpenFile(logPath, os.O_WRONLY|os.O_APPEND|os.O_CREATE, perms)
 	if err != nil {
 		return nil, err
 	}
 
+	if logConfig != nil {
+		size, err := f.Seek(0, os.SEEK_END)
+		if err != nil {
+			return nil, err
+		}
+		currentsize = uint64(size)
+		if maxsizeString, ok := logConfig["max-size"]; ok {
```

Comment:
just confirm, max-size don't have default value? what if max-size not config by user?

Reply:
👍

Reply:
If "max-size" isn't configured,   maxsize will not be set because this key is checked by "ok".
So, maxsize (uinit64) has a default value 0